modules/KalturaSupport/resources/mw.KCuePoints.js

```diff
@@ -405,6 +409,36 @@
 			}
 			return foundAttr;
 		},
+		/**
+		 * if have same CP earlier - hide current CuePoint
+		 * @param  allCP - array of CP where try to find same CP
+		 * @param  currentCuePointIndex - position from current CP
+		 *
+		 */
+		validateSameCuePoints:function (allCP, currentCuePointIndex) {
+			var currentCP = allCP[currentCuePointIndex];
+			var prevCP = this.getPrevCPWithCorrectType(allCP,currentCuePointIndex-1);
+			if(prevCP !== false && currentCP){
+				var differentByCreatedAt = Math.abs(currentCP.createdAt - prevCP.createdAt);
+				var differentByStartTime = Math.abs(currentCP.startTime - prevCP.startTime);
+				var isTheSamePartnerData = currentCP.partnerData === prevCP.partnerData;
+				var isTheSameTags = currentCP.tags === prevCP.tags;
+				if(isTheSamePartnerData && isTheSameTags && (differentByCreatedAt <= 3000 || differentByStartTime <= 3000)){
+					return false;
+				}
+			}
+			return true;
+		},
+		getPrevCPWithCorrectType: function (allCP,currentCuePointIndex) {
+			var prevCP = false;
+			for(var i = currentCuePointIndex; i>=0;i--){
+				if(allCP[i].cuePointType === "thumbCuePoint.Thumb"){
```

Comment:
@MaksymKorniiov 
why not if(allCP[i].cuePointType === currentCuePointIndex.cuePointType) ?

Reply:
@MaksymKorniiov I see that I told you via the ticket to use cue-point with type thumb 
Let's apply the comment above anyway :)